src/js/components/List/index.d.ts

```diff
@@ -43,7 +43,11 @@ export interface ListProps<ListItemType> {
     | string[]
     | { light: string | string[]; dark: string | string[] };
   border?: BorderType;
-  children?: (...args: any[]) => any;
+  children?: (
+    item: ListItemType,
+    index: number,
+    state?: { active: boolean },
+  ) => any;
```

Comment:
I am getting a typescript error on the `active` prop if I try to do something like this: `(item, index, {active}) => {}`.

Reply:
Ah, I misread the documentation - I thought the entire state object was optional rather than just the properties within.  Fix incoming!

Reply:
Just ran into an issue - I modified the test to validate and it seems that it's actually the entire state object that is optional.  Pinging on Slack for follow-up semi-synchronous conversation and will report back the decision for posterity's sake.

![image](https://github.com/grommet/grommet/assets/28069638/ce7e6821-9d31-4fc5-a0f5-a56007d13b86)

Reply:
Discussed offline, we are going to hold off on trying to support destructuring the active prop in the typescript definition. I think we can move forward with the types that are defined in this PR.